client/app/pages/dashboards/DashboardPage.jsx

```diff
@@ -0,0 +1,383 @@
+import React, { useState, useEffect } from 'react';
+import PropTypes from 'prop-types';
+import cx from 'classnames';
+import { map, isEmpty } from 'lodash';
+import { react2angular } from 'react2angular';
+import Button from 'antd/lib/button';
+import Checkbox from 'antd/lib/checkbox';
+import Dropdown from 'antd/lib/dropdown';
+import Menu from 'antd/lib/menu';
+import Icon from 'antd/lib/icon';
+import Modal from 'antd/lib/modal';
+import Tooltip from 'antd/lib/tooltip';
+import DashboardGrid from '@/components/dashboards/DashboardGrid';
+import { FavoritesControl } from '@/components/FavoritesControl';
+import { EditInPlace } from '@/components/EditInPlace';
+import { DashboardTagsControl } from '@/components/tags-control/TagsControl';
+import { Parameters } from '@/components/Parameters';
+import Filters from '@/components/Filters';
+import { Dashboard } from '@/services/dashboard';
+import recordEvent from '@/services/recordEvent';
+import { $route } from '@/services/ng';
+import getTags from '@/services/getTags';
+import { clientConfig } from '@/services/auth';
+import { durationHumanize } from '@/filters';
+import PromiseRejectionError from '@/lib/promise-rejection-error';
+import useDashboard, { DashboardStatusEnum } from './useDashboard';
+
+import './DashboardPage.less';
+
+function getDashboardTags() {
+  return getTags('api/dashboards/tags').then(tags => map(tags, t => t.name));
+}
+
+function buttonType(value) {
+  return value ? 'primary' : 'default';
+}
+
+function DashboardPageTitle({ dashboardOptions }) {
+  const { dashboard, canEditDashboard, updateDashboard, editingLayout } = dashboardOptions;
+  return (
+    <div className="page-title col-xs-8 col-sm-7 col-lg-7 p-l-0">
+      <FavoritesControl item={dashboard} />
+      <h3>
+        <EditInPlace
+          isEditable={editingLayout}
+          onDone={name => updateDashboard({ name })}
+          value={dashboard.name}
+          editor="input"
+          ignoreBlanks
+        />
+      </h3>
+      <img src={dashboard.user.profile_image_url} className="profile-image" alt={dashboard.user.name} />
+      <DashboardTagsControl
+        className="hidden-xs"
+        tags={dashboard.tags}
+        isDraft={dashboard.is_draft}
+        isArchived={dashboard.is_archived}
+        canEdit={canEditDashboard}
+        getAvailableTags={getDashboardTags}
+        onEdit={tags => updateDashboard({ tags })}
+      />
+    </div>
+  );
+}
+
+DashboardPageTitle.propTypes = {
+  dashboardOptions: PropTypes.object.isRequired, // eslint-disable-line react/forbid-prop-types
+};
+
+function RefreshButton({ dashboardOptions }) {
+  const { refreshRate, setRefreshRate, refreshing, refreshDashboard } = dashboardOptions;
+  const refreshRateOptions = clientConfig.dashboardRefreshIntervals;
```

Comment:
It seems to be missing the use of a policy to define allowed refresh intervals:

https://github.com/getredash/redash/blob/263305214ebe8d3cca903579ae96d637154a6fd4/client/app/pages/dashboards/dashboard.js#L116-L121

Reply:
:thinking:, currently it's still possible to put any interval you want through the url, should it be reflected in there?

Reply:
I think it was possible before as well, but we can also make sure that the value in the url is not lower than the minimum value the policy allows. Something like:

```
const refreshRate = max(urlValue, lowestPolicyValue);
```

Reply:
Yeah, my currently was meant to be `master` version, but cool, will go with it :+1: